.gitignore

```diff
@@ -1,5 +1,4 @@
 doc/*.html
-__pycache__
```

Comment:
Are you sure that you want to remove this?  (Won't this adversely affect human developers?)

Reply:
For whatever reason, it seems like `git` ignores them without adding it explicitly.  At least, with this in place I have not seen those "untracked" files.

Reply:
That's interesting; a search for .gitignore and __pycache__ shows lots of recommended `.gitignore` templates that include `__pycache__`. I don't see anything indicating that it should be ignored automatically. I wonder what's going on?

Reply:
So, so why it doesn't show up is because empty directories don't show up.  But if `.pyc` files get created in the tree that DO show up.  Which is what we want, because that should not happen.

Reply:
I certainly agree with looking for files in the tree use `git status --ignored`, etc.

Nobody should be using `git add .`, or `git add *`, or without being really sure.  There are lots of files that we _could_ to `.gitignore` to avoid things, but the `__pycache__` files are not one of them.

The creation of these files does not come from normal development tasks, or otherwise.  They come from somebody doing something wrong (at least with this commit) in how the python code was executed in the tree.

So _if_ they get created they should not be hidden, but should be front and center so that folks don't continue to use them.

Perhaps a `pre-add` git hook would help here.

Reply:
> _if_ they get created they should not be hidden

They aren't hidden:  they are visible via `ls`, `find`, et al.  `git status`, particularly without the `--ignored` switch, is simply the wrong tool for that job.

And, sure, a Git hook would help, but we would be setting it up to do the job that the `.gitignore` file is supposed to do up front.

> So, so why it doesn't show up is because empty directories don't show up. But if `.pyc` files get created in the tree that DO show up.

I'm not quite sure what you're trying to say, here.  If you want to propose a change to the `.gitignore` file to instead ignore `.pyc` files, I'm good with that.  If we need to add a trailing slash to the existing `__pycache__` entry to allow it to be properly treated as a directory, I'm good with that, too.  But, I really think we should have Git ignore the Python cache files.